### PR TITLE
Fix clippy error

### PR DESCRIPTION
**Pull request by chansuke**
Fix clippy error.

### CODE REVIEW

File: src/cli/self_update.rs [hunk 1]
```diff
@@ -89,7 +89,7 @@ pub(crate) const NEVER_SELF_UPDATE: bool = true;
 #[cfg(not(feature = "no-self-update"))]
 pub(crate) const NEVER_SELF_UPDATE: bool = false;
 
-#[derive(Clone, Debug, PartialEq)]
+#[derive(Clone, Debug, Eq, PartialEq)]
```
Comment (0xPoe) [inline src/cli/self_update.rs:92]: Which lint requires this change?

**Review by 0xPoe — COMMENTED** [review]
Thanks a lot!

If you can separate your changes with the lint name by commit, it would be great.

File: src/cli/self_update.rs (hunk 1 above)
Comment (chansuke) [reply src/cli/self_update.rs:92]: @hi-rustin Thank you so much for your review. I fixed [derive_partial_eq_without_eq](https://rust-lang.github.io/rust-clippy/master/index.html#derive_partial_eq_without_eq) lint error.

**Comment by chansuke** [comment]
@hi-rustin I have updated the commit log with clippy name. Some of the messages are duplicated because `cargo clippy` doesn't show the error at once and I procedurally fixed the log:pray:

File: download/src/lib.rs [hunk 2]
```diff
@@ -1,5 +1,6 @@
 //! Easy file downloading
 #![deny(rust_2018_idioms)]
+#![allow(clippy::type_complexity)]
```
Comment (0xPoe) [inline download/src/lib.rs]: I prefer to warn it. We may find a better way to fix it in the future.

**Review by 0xPoe — COMMENTED** [review]
Could you please squash your first and third commits? I think they are from the same lint.
rest LGTM. Thanks again!

**Comment by 0xPoe** [comment]
Also, could you please rebase your base branch? It seems you still using the 2022 master branch.

File: download/src/lib.rs (hunk 2 above)
Comment (0xPoe) [reply download/src/lib.rs]: @chansuke Could you please remove this? Then we can merge this PR.

**Comment by chansuke** [comment]
Sorry for many time. I will fix this tonight(JST)

File: download/src/lib.rs [hunk 3]
```diff
@@ -47,12 +47,13 @@ fn download_with_backend(
     }
 }
 
+type EventResult<'a> = &'a dyn Fn(Event<'_>) -> Result<()>;
```
Comment (0xPoe) [inline download/src/lib.rs]: We had some discussions on this change before: https://github.com/rust-lang/rustup/pull/3096#discussion_r1005434248
```suggestion
type DownloadCallback<'a> = &'a dyn Fn(Event<'_>) -> Result<()>;
```

**Review by 0xPoe — APPROVED** [review]
Thanks again.  I'm sorry for this back-and-forth discussion. I should have said it all at once.
Because this is a nit change. I'll push it directly.

**Comment by chansuke** [comment]
No problem. Thank you so much for your help.

**Pull request merged**